### PR TITLE
Feature/VNX-365: FSM web page for linking

### PR DESCRIPTION
**Pull request by erinisbell**
In order to test this, the easiest way is to replace the `AuthRedirect.js` file with something like this: 
```class AuthRedirect extends PureComponent<Props> {

  render() {
    const { location, currentUser } = this.props

    if(currentUser == null && location.pathname !== "/fsm/signin") {
      return <Redirect to="/fsm/signin"/>
    }
    else if(currentUser != null && location.pathname === "/fsm/signin") {
      return <Redirect to="/fsm/jobs?serviceticket=12345A&customer=KW&location=Nashville&shareId=d3035d8f-e18f-52de-af00-f2e3697ab801&partner=xoeye&tags=t1%7Ct2%7Ct3"/>
    }
    else return null
  }
}

export default withRouter(AuthRedirect);```


which will load a url for testing / pulling query parameters for linking jobs. I know this uses a lot of job activity files which were already there but I was thinking when we clean this project up we can re-name and re-organize the files (unless we think we need to go ahead and do this now).

### CODE REVIEW

File: share.schema.graphql [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  #  the user who generated the share
+  expiresAt: AWSDateTime!
+  extID: ID!
+  id: ID!
+  isFullShare: Boolean
+  sharer: String!
+}
+
+type Mutation {
+  createExternalJobShare(jobShare: JobShareInput!): JobShare
```
Comment (matteby) [inline share.schema.graphql]: The schema and fragment don't match. The parameter is `jobShare` here and `input` in the gql fragment.

**Comment by matteby** [comment]
@erinisbell I tried this out and once I made the changes in #4 it updated the job by adding the fields.

Could you review #4?

Also, I noticed the following:

- [ ] Tags can be passed as array (vs. multiple tag parameters with the same name) so we need to handle that case. Currently, if an array of tags are passed it results in a single tag entry on the job. For example, this URI: http://login.xoidev.io/fsm?serviceticket=12345A&customer=KW&location=Nashville&shareId=d3035d8f-e18f-52de-af00-f2e3697ab801&partner=xoeye&tags=t1%7Ct2%7Ct3 resulted in a single tag with this value **t1|t2|t3**
- [ ] Once a job is linked it still appears in the list available to link. The AC state: _"User is presented with list of COMPLETED jobs that only have the Name fields"_
- [ ] Also, I see that the share URL being is being created with a random UUID generated here. To support legacy FSM integrations we'll want to construct the URL with the **partnerId** and **sharedId** being passed to us. It's probably easiest to talk through the details tomorrow.

File: src/Components/util/LegacyAuth.js [hunk 2]
```diff
@@ -2,50 +2,64 @@
...
-
-    try {
-      const response = await fetch(authURI, authOptions)
-      return response.ok
-    }
-    catch (error) {
-      console.log("Request failed", error)
-    }
-    return false
+        //const baseURI = "https://login.staging.xoeye.com"
```
Comment (matteby) [inline src/Components/util/LegacyAuth.js]: @erinisbell This doesn't look right. This is somehow trying to change this file back to the older version. Looking at the history I'm not certain why it's trying to do it.

File: src/Components/util/urlQueryAsObject.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// @flow
+
+export default function urlQueryAsObject(query: any) {
```
Comment (matteby) [inline src/Components/util/urlQueryAsObject.js]: Could we use a library here?

We've used `queryparse` at other places in the FSM redirect logic.

```
const querystring = require('querystring');

const query = querystring.parse("shareId=9c4f3089-78e9-5462-9ea6-278e347e60d3&secret=18e7A915e20B-8A0-93C108e4457A-dfC-18&serviceTicket=12345A&customer=KW&location=Nashville&tags=t1&tags=t2&tags=t3")
const redirectQuery = querystring.stringify({ customer: query.customer, location: query.location, tags: query.tags, serviceticket: query.serviceTicket, shareId: query.shareId, partner: TODO })
const redirectQueryEncoded = encodeURIComponent(redirectQuery)
```

That should be more battle tested in terms of different corner cases and less stuff for us to own and maintain.

File: src/Components/util/Share.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+function makeShareUrl(id) {
+  const baseShareURL = getBaseURL()
+  return `${baseShareURL}/?id=${id}`
+}
+
+class ShareHandler {
+  // eslint-disable-next-line class-methods-use-this
+  shareFullJob(job: Object, createJobShareLink: Function, urlParams: Object) {
+    const id = urlParams.partner + '---' + urlParams.shareId
```
Comment (matteby) [inline src/Components/util/Share.js]: I didn't see a share id that fit this pattern in DDB after linking. Is there anything else I need to do to test it out?

File: src/Components/util/LegacyAuth.js (hunk 2 above)
Comment (erinisbell) [reply src/Components/util/LegacyAuth.js]: Yeah, I can change this back to what is in development at the moment.

File: src/Components/util/Share.js (hunk 4 above)
Comment (erinisbell) [reply src/Components/util/Share.js]: You shouldn't need to do anything else. Let me look into this more.

File: src/Components/util/urlQueryAsObject.js (hunk 3 above)
Comment (erinisbell) [reply src/Components/util/urlQueryAsObject.js]: I think not using a library is a smarter idea because that is something that we do own and can make changes when we need to and we won't be adding more dependencies to our project. 

Additionally with a dependency there is an added concerns around security. I do believe the better choice would be to convert all existing uses of the `querystring` library over to use our own utility function for handling query strings.

Comment (matteby) [reply src/Components/util/urlQueryAsObject.js]: The `querystring` library is a core node.js library and query strings are a web standard. I wouldn't imagine there are too many scenarios where we would want to extend or modify it.

In addition to `Can we code it ourselves?`, I'd propose we also consider what it would take to properly test and the cost to track down and fix potential issues if we don't ship with test coverage that is equivalent what is provided in a library.

Since it is pure javascript there is nothing node specific and there are several ports to the browser to make that library available on NPM. They also port over the tests from the core node.js repo which are quite extensive: https://github.com/mathiasvr/querystring/blob/master/test/core-tests/test-querystring.js
